### ISSUE TITLE
Duplicating seems broken for VisualShaderNodeCustom

### ISSUE THREAD

**Issue by Chaosus** [godotengine/godot#42573]
**Godot version:**
3556bc48a1d357ff0109e2677c2e02b925a3e46b

**OS/device including version:**
Win10

**Issue description:**
`duplicate()` function seems broken for `VisualShaderNodeCustom` in current master and leads to editor crash. This happens if the user using duplicate or copy/paste.
On the following screenshot duplicate() return zero however, it should return a duplicated reference:

![image](https://user-images.githubusercontent.com/3036176/95084125-1eb0ff80-0726-11eb-8ef0-6da9fe8394fb.png)
while the original node is correctly received and can be placed on the graph:

![image](https://user-images.githubusercontent.com/3036176/95084391-7a7b8880-0726-11eb-8c02-6640dbfd3c6e.png)

![vs_crash](https://user-images.githubusercontent.com/3036176/95085058-5ec4b200-0727-11eb-868b-568db7e4a677.gif)

I'm not sure whether this is related to the visual shader or the recent core changes. I have trouble understanding what's wrong with it - so I asking anyone to help.

**Minimal reproduction project:**
[TestShader.zip](https://github.com/godotengine/godot/files/5327614/TestShader.zip)

**Comment by pouleyKetchoupp** [comment]
Looks like a bug with gdscript.

The newly created resource gets unreferenced more than it gets referenced in the call to `set_script`, which causes the cast to `Ref<Resource>` to be null at the end of `duplicate`.

This is the callstack when the resource's ref count drops to 0:
```
 godot.windows.tools.64.exe!_atomic_decrement_impl(volatile unsigned int * pw) Line 69	C++
 godot.windows.tools.64.exe!atomic_decrement(volatile unsigned int * pw) Line 120	C++
 godot.windows.tools.64.exe!SafeRefCount::unrefval() Line 191	C++
 godot.windows.tools.64.exe!Reference::unreference() Line 78	C++
 godot.windows.tools.64.exe!Variant::clear() Line 1307	C++
 godot.windows.tools.64.exe!Variant::~Variant() Line 476	C++
 godot.windows.tools.64.exe!GDScriptFunction::call(GDScriptInstance * p_instance, const Variant * * p_args, int p_argcount, Callable::CallError & r_err, GDScriptFunction::CallState * p_state) Line 1590	C++
godot.windows.tools.64.exe!GDScript::_super_implicit_constructor(GDScript * p_script, GDScriptInstance * p_instance, Callable::CallError & r_error) Line 94	C++
 godot.windows.tools.64.exe!GDScript::_create_instance(const Variant * * p_args, int p_argcount, Object * p_owner, bool p_isref, Callable::CallError & r_error) Line 121	C++
 godot.windows.tools.64.exe!GDScript::instance_create(Object * p_this) Line 335	C++
 godot.windows.tools.64.exe!Object::set_script(const Variant & p_script) Line 869	C++
 godot.windows.tools.64.exe!Object::set(const StringName & p_name, const Variant & p_value, bool * r_valid) Line 401	C++
 godot.windows.tools.64.exe!Resource::duplicate(bool p_subresources) Line 229	C++
 godot.windows.tools.64.exe!VisualShaderEditor::_dup_paste_nodes(int p_type, int p_pasted_type, List<int,DefaultAllocator> & r_nodes, Set<int,Comparator<int>,DefaultAllocator> & r_excluded, const Vector2 & p_offset, bool p_select) Line 2209	C++
```

**Comment by Chaosus** [comment]
Seems like it is fixed (as b24c24f64b4ac9c30dc5a9e097163cbc64939645).
